pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeFactory.java

```diff
@@ -55,7 +56,11 @@ Runtime createContainer(
 
     default void doAdmissionChecks(Function.FunctionDetails functionDetails) { }
 
-    default Optional<FunctionAuthProvider> getAuthProvider() {
+    default Optional<? extends FunctionAuthProvider> getAuthProvider() {
```

Comment:
This change allows implementations of this interface to provide a more specific FunctionAuthProvider, which reduces the need for casting